h/h_api/bulk_api/observer.py

```diff
@@ -0,0 +1,40 @@
+"""Implementations of an 'Observer' which is informed of commands."""
+
+
+import json
+
+from h.h_api.enums import CommandStatus
+
+
+class Observer:
+    """A callback for being informed of every command processed by BulkJob.
+
+    This implementation serves as the base class and takes no action when
+    informed of a command."""
+
+    def observe_command(self, command, status):
+        """A call back for various stages in the command life cycle.
+
+        :param command: Command being inspected
+        :param status: Status of the command
+        """
+        pass
```

Comment:
If this were `__call__` then I think an observer could be an object-instance of a class with a `__call__` or it could just be a method or a function

Reply:
Yeah, but it's going to grow an `observer_result()` later on for catching the return values.